interface/main/tabs/js/tabs_view_model.js

```diff
@@ -19,15 +19,20 @@ function tabStatus(title,url,name,closable,visible,locked)
     self.title=ko.observable(title);
     self.url=ko.observable(url);
     self.name=ko.observable(name);
+    self.loading_text=ko.observable("Loading " + label + "...");
+    self.loader = ko.observable(true);
+    self.title.subscribe(function() {
+        self.loader(false);
+    });
     self.window=null;
     return this;
 }
 
 function tabs_view_model()
 {
     this.tabsList=ko.observableArray();
-    this.tabsList.push(new tabStatus("Loading <a style='font-size:80%'><i class='fa fa-spinner fa-pulse'></i></a>",webroot_url+"/interface/main/main_info.php","cal",true,true,false));
-    this.tabsList.push(new tabStatus("Loading <a style='font-size:80%'><i class='fa fa-spinner fa-pulse'></i></a>",webroot_url+"/interface/main/messages/messages.php?form_active=1","msg",true,false,false));
+    this.tabsList.push(new tabStatus("Loading <a style='font-size:80%'><i class='fa fa-spinner fa-pulse'></i></a>",webroot_url+"/interface/main/main_info.php","cal","Calender",true,true,false));
+    this.tabsList.push(new tabStatus("Loading <a style='font-size:80%'><i class='fa fa-spinner fa-pulse'></i></a>",webroot_url+"/interface/main/messages/messages.php?form_active=1","Messages","msg",true,false,false));
```

Comment:
These default labels are fine if user hasn't changed in globals for first and second tab.
Without having to parse those globals via top/main/jsGlobals either leave default labels empty or make them something like Default First Tab/Default Second Tab

Reply:
Should i make an ajax call to find globals and then pass label depending on them ? or is there already any function like that which i am unaware of?

Reply:
you could use library/ajax/phpvars_to_js.php to do the translations under top context.
this is auto called when main.php loads via jsFetchGlobals('top') around L-233.
Will still most likely run up against timing problems though.

Better off adding the static translations in main.php global js var's like several others.

Reply:
`jsFetchGlobals` also returns promise and it gets loded after `tabStatus` which leads to undefined  global vars 😕